data-network/generator/generator.py

```diff
@@ -537,43 +537,17 @@ def _natural_keys(self, text):
         return [self._atoi(c) for c in re.split(r"(\d+)", text)]
 
 
-if __name__ == "__main__":
+import argparse
 
-    if len(sys.argv) == 2:
-        print("Generate IPC instances")
-        filename = sys.argv[1]
-        seed = 0
-        for num_layers in range(2, 6):
-            for num_data_items in range(num_layers + 2, num_layers * 10, num_layers):
-                for num_scripts in range(
-                    num_data_items + 2, num_data_items * 3, num_data_items // 2
-                ):
+parser = argparse.ArgumentParser()
+parser.add_argument("data", type=int, help="the number of data items")
+parser.add_argument("layers", type=int, help="the number of layers, must be smaller than the number of data items")
+parser.add_argument("scripts", type=int, help="the number of scripts, must be larger or equal than [number of data items]-2")
+parser.add_argument("module", choices=["ring-network","small-network","tiny-network"],
+                    metavar="module",
+                    help='A module describing a network, one of %(choices)s. See example files in the source code directory.')
+parser.add_argument("--seed", type=int, default=42, help="random seed")
```

Comment:
I think the generator shouldn't become deterministic when no seed is given. IIRC, you simply need to change 42 to None. This should be sufficient for the Python random module. Not sure about the numpy random module.

Reply:
oh that is right